test/unit/specs/core/componentCreators.spec.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import componentCreators from "../../../../src/core/componentCreators";
+
+describe("componentCreators", () => {
+  it("is an array of component creators", () => {
+    expect(componentCreators).toEqual(jasmine.any(Array));
+    componentCreators.forEach(componentCreator => {
+      expect(componentCreator).toEqual(jasmine.any(Function));
+      expect(componentCreator.namespace).toEqual(jasmine.any(String));
+      expect(componentCreator.abbreviation).toEqual(jasmine.any(String));
+
+      if (componentCreator.configValidators) {
+        expect(componentCreator.configValidators).toEqual(jasmine.any(Object));
```

Comment:
We could do some more checks here on the configValidators
* check for only keys "defaultValue" & "validator"
* check that the validator function is of the correct name (We'd have to change newValidator in utils/configValidators/chain.js to be a named function)
* check that leaf nodes have at least "validator".
* check that the defaultValue validates (if it isn't null or undefined).
* make sure there aren't any overlapping fields (i.e. two components validating the same field)

Maybe put these in a separate spec file that would run it against the coreConfigValidators too.

Reply:
Created a JIRA issue: https://jira.corp.adobe.com/browse/CORE-35633